packages/signals_core/lib/src/core/value.dart

```diff
@@ -2,22 +2,31 @@ part of 'signals.dart';
 
 /// Signal that can be extended and used as a class
 class ValueSignal<T> extends _Signal<T> implements Signal<T> {
-  ValueSignal(super.value, {super.debugLabel});
+  ValueSignal(super.value, {super.debugLabel, super.equality});
 
   /// Should only be called to update the value of a signal if checks for equality
   /// have already been made.
   ///
   /// This is primarily used by the `ValueSignal` class to update the value
   /// since the reference is always the same.
-  void forceUpdate(T val) {
-    _updateValue(val);
+  void forceUpdate([T? val]) {
+    if (equality != null && val != null) {
```

Comment:
Please see the comment about a default Equality **if** possible.
equality != null everywhere in the codebase is sad :)